[BUG] rapids_scala213_nightly-dev-github build failure spark400

**Describe the bug**
rapids_scala213_nightly-dev-github build failed with: 

```shell
[2025-04-15T15:25:31.000Z] [ERROR] [Error] /tmp/20250415-88/shim/2/workspace/sql-plugin/src/main/scala/org/apache/spark/rapids/hybrid/HybridExecutionUtils.scala:452: wrong number of arguments for pattern org.apache.spark.sql.execution.datasources.LogicalRelation(relation: org.apache.spark.sql.sources.BaseRelation, output: Seq[org.apache.spark.sql.catalyst.expressions.AttributeReference], catalogTable: Option[org.apache.spark.sql.catalyst.catalog.CatalogTable], isStreaming: Boolean, stream: Option[org.apache.spark.sql.connector.read.streaming.SparkDataStream])
[2025-04-15T15:25:31.000Z] [ERROR] [Error] /tmp/20250415-88/shim/2/workspace/sql-plugin/src/main/scala/org/apache/spark/rapids/hybrid/HybridExecutionUtils.scala:453: local val newOptions in method applyOrElse is never used
[2025-04-15T15:25:31.001Z] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=unused-locals, site=org.apache.spark.rapids.hybrid.HybridExecOverrides.resolveHybridScanHint
[2025-04-15T15:25:31.001Z] [ERROR] [Error] /tmp/20250415-88/shim/2/workspace/sql-plugin/src/main/scala/org/apache/spark/rapids/hybrid/HybridExecutionUtils.scala:454: local val newRelation in method applyOrElse is never used
[2025-04-15T15:25:31.001Z] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=unused-locals, site=org.apache.spark.rapids.hybrid.HybridExecOverrides.resolveHybridScanHint
[2025-04-15T15:25:31.001Z] [ERROR] three errors found
``` 

**Steps/Code to reproduce bug**
Rerun nightly build

**Expected behavior**
Build passes

**Environment details (please complete the following information)**

**Additional context**
Possibly related to https://github.com/NVIDIA/spark-rapids/pull/12370 
cc: @sperlingxx

--- a/sql-plugin/src/main/scala/org/apache/spark/rapids/hybrid/HybridExecutionUtils.scala
+++ b/sql-plugin/src/main/scala/org/apache/spark/rapids/hybrid/HybridExecutionUtils.scala
@@ -449,10 +449,11 @@ object HybridExecOverrides extends Logging {
     plan.resolveOperatorsWithPruning(_.containsPattern(TreePattern.UNRESOLVED_HINT)) {
       case UnresolvedHint(n, Nil, child) if n.toUpperCase(Locale.ROOT).equals(HYBRID_SCAN_HINT) =>
         child.transformUp {
-          case op@LogicalRelation(rel: HadoopFsRelation, _, _, _) =>
-            val newOptions = rel.options.updated(HYBRID_SCAN_TAG, "")
-            val newRelation = rel.copy(options = newOptions)(rel.sparkSession)
-            op.copy(relation = newRelation)
+          case rel: LogicalRelation if rel.relation.isInstanceOf[HadoopFsRelation] =>
+            val hdfsRel = rel.relation.asInstanceOf[HadoopFsRelation]
+            val newOptions = hdfsRel.options.updated(HYBRID_SCAN_TAG, "")
+            val newRelation = hdfsRel.copy(options = newOptions)(hdfsRel.sparkSession)
+            rel.copy(relation = newRelation)
         }
     }
   }

